azure-samples/src/main/java/com/microsoft/azure/management/compute/samples/ParallelCreate.java

```diff
@@ -0,0 +1,341 @@
+                for (int i = 0; i < vmCount; i++) {
+                    VirtualMachine.DefinitionStages.WithCreate creatableVirtualMachine = azure.virtualMachines()
+                            .define("VM-" + i)
+                            .withRegion(Region.US_EAST)
+                            .withExistingResourceGroup(resourceGroup)
+                            .withNewPrimaryNetwork(creatableNetwork)
+                            .withPrimaryPrivateIpAddressDynamic()
+                            .withoutPrimaryPublicIpAddress()
+                            .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS)
+                            .withRootUserName("tirekicker")
+                            .withPassword("12NewPA$$w0rd!")
+                            .withSize(VirtualMachineSizeTypes.STANDARD_D3_V2)
+                            .withNewStorageAccount(creatableStorageAccount);
+                    creatableVirtualMachines1.add(creatableVirtualMachine);
+                }
+
+                System.out.println("Creating the virtual machines");
+                CreatedResources<VirtualMachine> virtualMachines1 = azure.virtualMachines().create(creatableVirtualMachines1);
+                System.out.println("Created virtual machines");
+                for (VirtualMachine virtualMachine : virtualMachines1) {
+                    System.out.println(virtualMachine.id());
+                }
+
+                // Create a virtual network with two Subnets – frontend and backend
+                //
+                Network network = azure.networks()
+                        .define(networkName2)
+                        .withRegion(Region.US_EAST)
+                        .withExistingResourceGroup(resourceGroup)
+                        .withAddressSpace("172.16.0.0/16")
+                        .defineSubnet("Front-end")
+                        .withAddressPrefix("172.16.1.0/24")
+                        .attach()
+                        .defineSubnet("Back-End")
+                        .withAddressPrefix("172.16.2.0/24")
+                        .attach()
+                        .create();
+
+                //============================================================
+                // Define a network security group for the front end of a subnet
+                // front end subnet contains two rules
+                // - ALLOW-SSH - allows SSH traffic into the front end subnet
+                // - ALLOW-WEB- allows HTTP traffic into the front end subnet
+
+                NetworkSecurityGroup.DefinitionStages.WithCreate frontEndNSGCreatable = azure.networkSecurityGroups()
+                        .define(frontEndNSGName)
+                        .withRegion(Region.US_EAST)
+                        .withNewResourceGroup(rgName)
+                        .defineRule("ALLOW-SSH")
+                        .allowInbound()
```

Comment:
for readability, a recommended tabbing convention would be to add an extra indent in front of child withers and go back to the parent's indentation after .attach()